CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs

```diff
@@ -151,6 +151,21 @@ public static partial class Extensions
             await botToUser.PostAsync(message, cancellationToken);
         }
 
+        /// <summary>
+        /// Ends the conversation.
+        /// </summary>
+        /// <param name="botToUser">Communication channel to use.</param>
+        /// <param name="code">The conversation end code.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns></returns>
+        public static async Task EndConversation(this IBotToUser botToUser, string code = EndOfConversationCodes.CompletedSuccessfully, CancellationToken cancellationToken = default(CancellationToken))

```

Comment:
EndConversation is a misleading name. Probably PostEndConversation or SayEndCoversation is a better name for this method, since it is not impacting the stack and only sends an EndOfConversation Activity to channel.